sbt-dotty/src/dotty/tools/sbtplugin/DottyIDEPlugin.scala

```diff
@@ -64,6 +63,11 @@ object DottyIDEPlugin extends AutoPlugin {
     Def.derive(projectConfig := {
       if (sources.value.isEmpty) None
       else {
+        // Not needed to generate the config, but this guarantees that the
+        // generated config is usable by an IDE without any extra compilation
+        // step.
+        val _ = compile.value
```

Comment:
Right, but all compileForIDE does is call `compile` on each dotty configuration, here we already are in a task run in every dotty configuration, so `compile` is fine.